scipy/signal/tests/test_signaltools.py

```diff
@@ -1093,19 +1093,13 @@ def test_types(self, dtype):
         assert_equal(signal.medfilt(in_typed).dtype, dtype)
         assert_equal(signal.medfilt2d(in_typed).dtype, dtype)
 
-    def test_types_deprecated(self):
-        dtype = np.longdouble
-        in_typed = np.array(self.IN, dtype=dtype)
-        msg = "Using medfilt with arrays of dtype"
-        with pytest.deprecated_call(match=msg):
-            assert_equal(signal.medfilt(in_typed).dtype, dtype)
-        with pytest.deprecated_call(match=msg):
-            assert_equal(signal.medfilt2d(in_typed).dtype, dtype)
-
-
     @pytest.mark.parametrize('dtype', [np.bool_, np.complex64, np.complex128,
-                                       np.clongdouble, np.float16,])
+                                       np.clongdouble, np.float16, np.object_,
+                                       "float96", "float128"])
     def test_invalid_dtypes(self, dtype):
+        if np.finfo(np.longdouble).dtype != dtype and dtype in ["float96", "float128"]:
+            pytest.skip(f"Platform does not support {dtype}")
```

Comment:
I find the ordering of the conditions hard to follow in terms of the logic behind. Perhaps a comment would help. In particular, the equality between a proper dtype and a sometimes-dtype-sometimes-string is weird, especially as it seems redundant with the `in` check afterwards.

If this was based on my mock [example](https://github.com/scipy/scipy/pull/19673#discussion_r1451617209), then I'd now write it as
```suggestion
        if dtype in ["float96", "float128"] and np.finfo(np.longdouble).dtype.itemsize < 12:
            pytest.skip(f"Platform does not support {dtype}")
```

Reply:
I will add a comment explaining the logic but unfortunately the suggestion won't work as a platform will only have float96 or float128 so we still need to skip one of them even if `np.finfo(np.longdouble).dtype.itemsize > 12`